src/components/android-settings-container.js

```diff
@@ -74,12 +85,20 @@ class AndroidSettingsContainer extends React.Component {
             return true
         }
 
+        let { numberSmsToSend, numberSmsConfirmation } = this.state
```

Comment:
These are assigned from `state`, but when you re-assign them later (`numberSmsConfirmation = null`, for instance) you are NOT updating the value of `this.data.numberSmsConfirmation` but rather only the value of the variable `numberSmsConfirmation`.  Because you only use `this.state.numberSmsConfirmation` later, this assignment *won't actually work*

Reply:
Ohh I see, I forgot to use this assignment, it was supposed to be like this:
```
const androidData = {
            metadataSync: this.state.metadataSync,
            dataSync: this.state.dataSync,
            numberSmsToSend: numberSmsToSend,
            numberSmsConfirmation: numberSmsConfirmation,
            reservedValues: this.state.reservedValues,
            encryptDB: this.state.encryptDB,
            lastUpdated: new Date().toJSON(),
        }
```
but I probably removed that change before commit it, sorry I totally understand this was not working. So now I'm not using this: `let { numberSmsToSend, numberSmsConfirmation } = this.state`   instead I'm just re assigning the states